### PR TITLE
Introduce solidrun lx2160acex7

### PR DESCRIPTION
**Pull request by rehsack**
@rabeeh, @tingliu, @otavio, @zandrey - please take a first look

@rabeeh - maybe you could give some hints wrt. which SERDES config is intended for which Solid-Run product.

### CODE REVIEW

**Comment by rehsack** [comment]
> This a a **HUGE** work! 😮
> 
> In general, there are a lot of patches that potentially should be looked at and picked up by NXP, but in general - it is OK.
> 
> I had couple of questions, they are below.

The work has been done by @rabeeh - I just took it from http://github.com/SolidRun/lx2160a_build and integrate them into kernel.

File: recipes-kernel/linux/linux-fslc-qoriq_5.4.bbappend [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FILESEXTRAPATHS_append_lx2160acex7 := "${THISDIR}/${PN}-lx2160acex7:"
+
+SRC_URI_append_lx2160acex7 = "\
+    file://0001-arm64-dts-lx2160a-add-lx2160acex7-device-tree-build.patch \
+    file://0002-arm64-dts-lx2160a-add-lx2160acex7-device-tree.patch \
+    file://0005-arm64-dts-lx2160a-cex7-add-ltc3882-support.patch \
+    file://0006-arm64-dts-lx2160a-cex7-add-on-module-eeproms.patch \
+    file://0007-pci-hotplug-declare-IDT-bridge-as-hotpluggabl-bridge.patch \
+    file://0008-pci-spr2803-quirk-to-fix-class-ID.patch \
+"
```
[the remark this answers is not in the document]
Comment (rehsack) [reply recipes-kernel/linux/linux-fslc-qoriq_5.4.bbappend:10]: I don't know - for the moment I thinks, it's a start. I added locally a linux-fslc-qoriq-solidrun branch, but I don't know whether I can maintain it longer ...
I could - of course - add a linux-fslc branch which is maintained by linux-fsl community.

File: wic/lx2160acex7.wks.in [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#         +-----------------------+
+#
+
+part RCW --source rawcopy --sourceparams="file=atf/bl2_auto.pbl" --ondisk mmcblk --no-table --align 4
+part FIP --source rawcopy --sourceparams="file=atf/fip_uboot.bin" --ondisk mmcblk --no-table --align 1024
+part DDR --source rawcopy --sourceparams="file=ddr-phy/fip_ddr_all.bin" --ondisk mmcblk --no-table --align 8192
+part MCFW --source rawcopy --sourceparams="file=mc_app/mc.itb" --ondisk mmcblk --no-table --align 10240
+part DPL --source rawcopy --sourceparams="file=mc-utils/dpl-eth.8x10g.19.dtb" --ondisk mmcblk --no-table --align 13312
+part DPC --source rawcopy --sourceparams="file=mc-utils/dpc-8_x_usxgmii.dtb" --ondisk mmcblk --no-table --align 14336
+part Kernel --source rawcopy --sourceparams="file=fitImage" --ondisk mmcblk --no-table --align 16384
```
[the remark this answers is not in the document]
Comment (rehsack) [reply wic/lx2160acex7.wks.in:61]: The fitimage is likely for booting from SPI flash, I don't catch it up for sd/emmc boots.

**Comment by rehsack** [comment]
With today's update, normal distro_boot works with the layout defined in .wks file

**Comment by otavio** [comment]
@rehsack there are fixes we can send to meta-freescale?

**Comment by rehsack** [comment]
@otavio: as far as I can tell, not directly. Maybe the uboot-script, but let's see how it evolves.
I plan to do a similar build as I did for lx2160a-cex7 for ls2088a-rdb, lx2160a-rdb and ls2080a-rdb ... but this is +/- 3..4 weeks in future.

**Comment by rehsack** [comment]
@otavio - shall I add an "Upstream-Status: inappropriate" to all patches?

**Comment by otavio** [comment]
Yes, I think so. 

One thing we might consider is to provide +fslc forks for some of those components with those improvements and use them for all board. Does it makes sense?

**Comment by rehsack** [comment]
I need more practical experience to cherry-pick. At the very moment I'd say: it's lx2160a-cex7 only. When I got more experience with fsl-lsch3 platform, I can re-evaluate.
Dunno whether @rabeeh or @ting-liu see some common improvements now.

**Comment by rehsack** [comment]
@otavio: Upstream-Status added

File: recipes-bsp/atf/atf_git.bbappend [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FILESEXTRAPATHS_append_lx2160acex7 := "${THISDIR}/${PN}-lx2160acex7:"
+
+SRC_URI_append_lx2160acex7 = "\
+	file://0001-plat-nxp-Add-lx2160acex7-module-support.patch \
+	file://0002-plat-nxp-lx2160a-auto-boot.patch \
+"
+
+# ddrphyopt_lx2160acex7 = "fip_ddr_sec"
```
Comment (otavio) [inline recipes-bsp/atf/atf_git.bbappend:8]: This is commented code, why?

File: wic/lx2160acex7.wks.in [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#         +-----------------------+
+#
+
+part RCW --source rawcopy --sourceparams="file=atf/bl2_auto.pbl" --ondisk mmcblk --no-table --align 4
+part FIP --source rawcopy --sourceparams="file=atf/fip_uboot.bin" --ondisk mmcblk --no-table --align 1024
+part DDR --source rawcopy --sourceparams="file=ddr-phy/fip_ddr_all.bin" --ondisk mmcblk --no-table --align 8192
+part MCFW --source rawcopy --sourceparams="file=mc_app/mc.itb" --ondisk mmcblk --no-table --align 10240
+part DPL --source rawcopy --sourceparams="file=mc-utils/dpl-eth.8x10g.19.dtb" --ondisk mmcblk --no-table --align 13312
+part DPC --source rawcopy --sourceparams="file=mc-utils/dpc-8_x_usxgmii.dtb" --ondisk mmcblk --no-table --align 14336
+# part Kernel --source rawcopy --sourceparams="file=fitImage" --ondisk mmcblk --no-table --align 16384
```
Comment (otavio) [inline wic/lx2160acex7.wks.in:61]: Please don't do commented code.

File: recipes-bsp/atf/atf_git.bbappend (hunk 3 above)
Comment (rehsack) [reply recipes-bsp/atf/atf_git.bbappend]: I added this very early and forgot to cleanup. Maybe this becomes necessary again when using secure-boot.

File: wic/lx2160acex7.wks.in (hunk 4 above)
Comment (rehsack) [reply wic/lx2160acex7.wks.in]: I left this accidentally while thinking how to put this into the NOR-flash, removed.

**Pull request merged**